File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        lst = []
        
        def helper(root, lst):
            if not root:
                return
            
            helper(root.left, lst)
            lst.append(root.val)
            helper(root.right, lst)
        
        helper(root, lst)
        return lst
        
            
                
                
        
        

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.c
void helper_inorder(struct TreeNode* root, int* arr, int* counter) {
     if (!root) return;
     else{
         helper_inorder(root->left, arr, counter);
         arr[*(counter)] = root->val;
         *counter += 1;
         helper_inorder(root->right, arr, counter);
     }

 }
 int* inorderTraversal(struct TreeNode* root, int* returnSize) {
     int* arr = (int*)malloc(100 * sizeof(int));
     int counter = 0;

     helper_inorder(root, arr, &counter);

     *returnSize = counter;
     arr = (int*)realloc(arr, counter * sizeof(int));
     return arr;
 }
